Automattic/wp-calypso
# Split out the rest of lib/analytics

This PR concludes the modularisation of `lib/analytics`, by splitting out `page-view`, `refer`, and `tracks` into their own modules. `lib/analytics` will no longer have an index file after this PR, so individual modules will need to be imported.

The PR sadly doesn't seem to come with any reductions in bundle size, since all of the extracted modules are needed by the analytics middleware. I did experiment with lazy loading in the analytics middleware, but the bundle savings didn't appear to justify the added complexity.

Note that there will be linting errors due to existing problems on some of the files this PR touches, which I deemed to be out of scope to fix here.

#### Changes proposed in this Pull Request

* Rename `lib/analytics/refer.js` to `lib/analytics/track-affiliate-referral.js`
* Split `page-view`, `refer`, and `tracks` into their own modules
* Remove `lib/analytics/index.js`
* Update all code accordingly (quite a few files)

#### Testing instructions

This PR mostly just moves things around, so the build, unit tests, and E2E tests should cover most of the ground here. Testing out the basic functionality of `page-view`, `refer`, and `tracks` to ensure it didn't get broken should cover the rest.
